lib/elastic_apm/timed_worker.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  # @api private
+  class TimedWorker
+    include Log
+
+    SLEEP_INTERVAL = 0.1
+
+    # @api private
+    class StopMsg
+    end
+
+    # @api private
+    class RequestMsg
+      def initialize(path, payload)
+        @path = path
+        @payload = payload
+      end
+
+      attr_reader :path, :payload
+    end
+
+    # @api private
+    class ErrorMsg
+      def initialize(error)
+        @error = error
+      end
+
+      attr_reader :error
+    end
+
+    def initialize(config, messages, batched_transactions, adapter)
+      @config = config
+      @adapter = adapter
+      @messages = messages
+      @batched_transactions = batched_transactions
+
+      @last_sent_transactions = Time.now.utc
+
+      @serializers = Struct.new(:transactions, :errors).new(
+        Serializers::Transactions.new(config),
+        Serializers::Errors.new(config)
+      )
+
+      @rand = rand(999)
+    end
+
+    attr_reader :config, :messages, :batched_transactions
+
+    def run_forever
+      loop do
+        run_once
+        sleep SLEEP_INTERVAL
+      end
+    end
+
+    def run_once
+      process_messages
+
+      flush_transactions if should_flush_transactions?
```

Comment:
I think the order of `process_messages` and `flush_transactions` should be changed, or `process_messages` should be called from `flush_transactions` at the end.